linera-core/src/client.rs

```diff
@@ -286,7 +286,7 @@ pub enum ChainClientError {
 
 impl From<Infallible> for ChainClientError {
     fn from(infallible: Infallible) -> Self {
-        infallible.into()
```

Comment:
oh noo this was so much more in the spirit of Curry-Howard

Reply:
Was it? :smile:
It would make sense to me if `From<Infallible> for T` were implemented for all `T` in the standard library (ex falso quodlibet). But it isn't: We're just doing infinite recursion here.

Reply:
I think EFQ isn't usually defined in terms of an infinite regression :)